scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,115 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def test_gttrf_gttrs():
+    np.random.seed(42)
+    for index, dtype in enumerate(DTYPES):
+        n = 10
+        if index % 2 == 0:
+            rtol = 10**-(np.finfo(dtype).precision-1)
+        else:
+            rtol = 10**-(np.finfo(dtype).precision-3)
+        if index < 2:
+            du = np.random.rand(n-1)
+            d = np.random.rand(n)
+            dl = np.random.rand(n-1)
+        else:
+            du = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+            d = np.random.rand(n) + np.random.rand(n) * 1j
+            dl = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+
+        diag_cpy = np.array([dl.copy(), d.copy(), du.copy()])
+
+        A = np.diag(d) + np.diag(dl, -1) + np.diag(du, 1)
+        x = np.random.rand(n)
+        b = A @ x
+
+        gttrf, gttrs = get_lapack_funcs(('gttrf', 'gttrs'), dtype=dtype)
+
+        _dl, _d, _du, du2, ipiv, info = gttrf(dl, d, du)
+
+        assert_array_equal(dl, diag_cpy[0])
+        assert_array_equal(d, diag_cpy[1])
+        assert_array_equal(du, diag_cpy[2])
+
+        perm_fnl = np.arange(n)
+        for i, piv in enumerate(ipiv):
+            perm_fnl[i], perm_fnl[piv-1] = perm_fnl[piv-1], perm_fnl[i]
+
+        _A = np.diag(_dl, -1) + np.diag(_d, 0) + np.diag(_du, 1)
+        #assert_allclose(_A, A[perm_fnl], rtol=rtol)
+
+        b_cpy = b.copy()
+        x_gttrs, info = gttrs(_dl, _d, _du, du2, ipiv, b)
+        assert_array_equal(b, b_cpy)
+
+        assert_allclose(x, x_gttrs, rtol=rtol)
+
+        if index < 2:
+            dl_misshaped = np.random.rand(n)
+            d_misshaped = np.random.rand(n-1)
+            du_misshaped = np.random.rand(n)
+        else:
+            dl_misshaped = np.random.rand(n) + np.random.rand(n) * 1j
+            d_misshaped = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+            du_misshaped = np.random.rand(n) + np.random.rand(n) * 1j
+
+        with assert_raises(ValueError):
+            gttrf(dl_misshaped, d, du)
+        with assert_raises(ValueError):
+            gttrf(dl, d_misshaped, du)
+        with assert_raises(ValueError):
+            gttrf(dl, d, du_misshaped)
+
+        with assert_raises(Exception):
+            gttrf(dl[0], d[:1], du[0])
+
+        transpose_options = ["N", "C", "T"]
```

Comment:
We should probably just do "T" for real data types and "C" for complex data types.